### PR TITLE
Resolves #17

### LINKED ISSUE #17: Retain pubspec.yml contents
Hi there,

thanks for your awesome package! Is there a possibility to retain comments, linebreaks etc. when bumping the version? After bumping the version my pubspec file is very condensed and all comments and linebreaks are deleted.

### PR DESCRIPTION
**Pull request by Agraphie**
Resolves issue #17 by not using pubspec package for writing new version, but by searching for the version string in the pubspec file, incrementing it and writing back the string to the pubspec file.

### CODE REVIEW

File: test/functional_test.dart [hunk 1]
```diff
@@ -19,7 +19,7 @@ void main() {
 
   setUp(() async {
     temp = await Directory.systemTemp.createTemp();
-    File('test/pubspec_sample.yaml').copy('${temp.path}/pubspec.yaml');
+    await File('test/pubspec_sample.yaml').copy('${temp.path}/pubspec.yaml');
```
Comment (f3ath) [inline test/functional_test.dart]: good catch, thanks

File: lib/src/update_version.dart [hunk 2]
```diff
@@ -13,10 +13,45 @@ abstract class UpdateVersion extends Command {
...
+    Version newVersion = nextVersion(oldVersion);
+    console.log(newVersion.toString());
+
+    _writeNewPubSpec(pubSpec, oldVersion, newVersion);
+  }
+
+  Future _writeNewPubSpec(
+      File pubSpec, Version oldVersion, Version newVersion) async {
+    String newPubSpec = pubSpec.readAsStringSync().replaceFirst(
+        'version: "${oldVersion}"', "version: " + '"${newVersion.toString()}"');
```
Comment (f3ath) [inline lib/src/update_version.dart]: What if a different quotation is used?
```yaml
version: '1.2.3+4'
```
or
```yaml
version: 1.2.3+4
```

**Comment by f3ath** [comment]
Thanks for the PR. I'd like to see the pubspec manipulation logic extracted to a separate class. This way we could you it also for reading the version and get rid of the entire `pubspec` dependency. Also please consider formatting the code, it will make the CI happy.

File: lib/src/update_version.dart (hunk 2 above)
Comment (Agraphie) [reply lib/src/update_version.dart]: True, thanks. The ouput shouldn't matter but for the input I guess it's better to use a simple regex. I'll add it.

**Comment by Agraphie** [comment]
Good idea. I'll do that. I'll (possibly) also try to add this feature further down in the pubspec package or so. But after a quick glance I didn't find an easy way. In any case, if it gets added further down the pipe, we can just remove the code here again and revert to `copy(...).save()`. What do you think? Also I'm quite the beginner in Dart, so any feedback is appreciated :D

**Comment by f3ath** [comment]
Sure. Open another PR to [this branch](https://github.com/f3ath/pubspec-version/tree/retain-yaml-formatting), I'll help with refactoring.

**Pull request closed without merging**